x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1652,13 +1690,15 @@ public boolean test(String profile, InetSocketAddress peerAddress) {
                 tracer,
                 new TLSConfig(sslConfiguration, sslService::createSSLEngine),
                 acceptPredicate,
-                Netty4HttpHeaderValidator.NOOP_VALIDATOR
+                new HttpHeadersValidator(authenticateMessage)
```

Comment:
This is inside the "Security plugin land".
Here, there's code that runs authentication, that is hooked into code that netty code exposes for generic request validation.